src/main/java/com/redhat/quarkus/mandrel/collector/report/model/ImageStatsCollection.java

```diff
@@ -200,3 +255,40 @@ public RunnerInfo deleteRunnerInfo(long id) {
         return r;
     }
 }
+
+class DateRange {
+    final Date start;
+    final Date end;
+
+    private DateRange(Date start, Date end) {
+        this.start = start;
+        this.end = end;
+    }
+
+    /**
+     * Parse the start and end date strings and return a DateRange object.
+     * If one of the dates is null, it will be set to the minimum or maximum date accordingly.
+     */
+    static DateRange fromString(String start, String end) {
+        assert start != null || end != null;
+        if (start == null) {
+            start = "1970-01-01 00:00:00.000";
+        }
+        if (end == null) {
+            end = LocalDateTime.now().format(ImageStatsCollection.CREATED_DATE_FORMATTER);
+        }
```

Comment:
We seem to be about two minds about this. 1.) we fail the assert if both are `null`. 2) with assertions turned off the code runs fine. I'd suggest to decide which it should be and enforce that (irrespective of assertions turned on or off).